button.cpp

```diff
@@ -75,7 +75,11 @@ void myButton::setPressedAlwaysTrue(){
 
 
 void myButton::setEvents(ALLEGRO_EVENT *ev){
-	mouse_event_input(ev);
+
+}
+
+void myButton::update_input(ALLEGRO_EVENT *e){

```

Comment:
Você criou esse metodo update_input() para fazer a mesma coisa que o setEvent() faz? Se esse metodo novo fizer algo diferente está ok. Se ele for a mesma coisa, apenas um nome melhor, isso também é OK... mas dai vc tem que deletar o setEvent() pra não ficar codigo morto no programa. Também tem que entrar em interfaceComponent.h e renomear setEvents() para update_input(). Eu também adicionaria o metodo draw() que você criou dentro do interfaceComponent.h. Isso obriga todo mundo a implementar ele.